modules/ROOT/pages/expressions/predicates/label-expression-predicates.adoc

```diff
@@ -0,0 +1,249 @@
+
+| "Alice" | true
+| "Cecil" | true
+| "Cecilia" | true
+| "Charlie" | true
+| "Daniel" | false
+
+2+d|Rows: 5
+|===
+
+When '<exp>', then the label expression predicate results in `null`, e.g. if `p` is `null`, then `p:!CEO` results in `null`.
+
+[source, cypher]
+----
+MATCH (p:Person)
+OPTIONAL MATCH (p)-[r]->(m)
+RETURN p.name AS name,
+       m:!CEO AS doesNotWorkDirectlyForTheCEO
+----
+
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| name | doesNotWorkDirectlyForTheCEO
+
+| "Alice" | true
+| "Cecil" | true
+| "Cecilia" | true
+| "Charlie" | true
+| "Daniel" | false
+| "Eskil" | null
+
+2+d|Rows: 6
+|===
+
+The function xref:functions/scalar.adoc#functions-coalesce[coalesce] allows to turn a `null` into a default values:
+
+[source, cypher]
+----
+MATCH (p:Person)
+OPTIONAL MATCH (p)-[r]->(m)
+RETURN p.name AS name,
+       coalesce(m:!CEO, false) AS doesNotWorkDirectlyForTheCEO
+----
+
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| name | doesNotWorkDirectlyForTheCEO
+
+| "Alice" | true
+| "Cecil" | true
+| "Cecilia" | true
+| "Charlie" | true
+| "Daniel" | false
+| "Eskil" | false
+
+2+d|Rows: 6
+|===
+
+[role=label--new-Neo4j-2025.07 label--cypher-25-only]
+[[dynamic-node-label-expression-predicate]]
+== Test that node matches dynamic label expression
+
+If we do not know the manager labels, we can infer a list of `managerLabels` by collecting all non-`Person` labels from people with a direct report.
+Nodes can be tested with the label expression predicate to match xref:patterns/reference.adoc#dynamic-label-and-type-expressions[dynamic label expression].
+`p:$any(managerLabels)` test whether node `p` has at least on of the label in the list `managerLabels`.
+
+[source, cypher]
+----
+CYPHER 25
+MATCH ()-[r]->(p:Person)
+UNWIND labels(p) AS label
+FILTER label <> "Person"
+RETURN COLLECT(label) AS managerLabels
+NEXT
+MATCH (p)
+RETURN p.name AS name, p:$any(managerLabels) AS isManager
+----
+
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| name | isManager
+
+| "Alice" | true
+| "Cecil" | false
+| "Cecilia" | false
+| "Charlie" | false
+| "Daniel" | true
+| "Eskil" | true
+
+2+d|Rows: 6
+|===
+
+[node-label-expression-predicate]
+== Test that relationship has a certain relationship type
+
+Given `r` is a relationship, `r:WORKS_FOR` tests whether `r` has the relationship type `WORKS_FOR` or not and result in `true` or `false`, respectively.
+
+// tag::relationship-label-expression-predicate[]
+[source, cypher]
+----
+MATCH (p:Person)-[r]->()
+RETURN p.name AS name,
+       r:WORKS_FOR AS isNotManager
+----
+// end::relationship-label-expression-predicate[]
+
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| name | isNotManager
+
+| "Alice" | false
+| "Cecil" | true
+| "Cecilia" | true
+| "Charlie" | true
+| "Daniel" | false
+
+2+d|Rows: 5
+|===
+
+If `r` is `null`, then the label expression predicate, e.g. `r:WORKS_FOR|REPORTS_TO` results in `null`.
```

Comment:
Perhaps you should add the sentence about using coalesce to change nulls to default values here as well. Having read the whole page it is clear, but if someone is looking at this example only it might be a bit confusing that this sentence mentions null and the result below does not